taca/illumina/Runs.py

```diff
@@ -703,6 +703,31 @@ def _process_simple_lane_with_single_demux(
                 os.unlink(dest)
         os.symlink(source, dest)
 
+        # Replace the file laneBarcode.html
+        source_html_report_laneBarcode = os.path.join(
+            source, "html", self.flowcell_id, "all", "all", "all", "laneBarcode.html"
+        )
+        dest_html_report_laneBarcode = os.path.join(
+            dest, "html", self.flowcell_id, "all", "all", "all", "laneBarcode.html"
+        )
+        if os.path.exists(source_html_report_laneBarcode) and os.path.exists(
+            dest_html_report_laneBarcode
+        ):
+            html_report_laneBarcode_parser = LaneBarcodeParser(
+                source_html_report_laneBarcode
+            )
+            # Remove the trailing "_SX" postfix from samples names for BCL Convert when it handles SmartSeq3 libraries
+            for entry in html_report_laneBarcode_parser.sample_data:
+                if "_S" in entry["Sample"]:
+                    entry["Sample"] = "_".join(entry["Sample"].split("_")[:2])
+            html_report_laneBarcode_parser.sample_data = sorted(
+                html_report_laneBarcode_parser.sample_data,
+                key=lambda k: (k["Lane"].lower(), k["Sample"]),
+            )
+            _generate_lane_html(
```

Comment:
I'm not entirely sure how this works, but are we overwriting the file that exists at `dest_html_report_laneBarcode` here?

Reply:
Or is it the same file here since we symlink it `os.symlink(source, dest)`?

Reply:
Yes, we are overwriting the file. Otherwise we will have to create hard links of all files under `source`  to `dest` individually, in which case the scripts will be much more complicated...